controller/service.go

```diff
@@ -98,13 +106,21 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 		}
 	}
 
+	// Check for any deprecated annotations
+	for key := range svc.Annotations {
```

Comment:
if we put this here, we'll generate an event (for every deprecated annotation) each time we receive an update of the service or the configuration changes (this is because we call this function on every service update, or config change, to make sure the allocation is still valid), and we might risk to flood the cluster.

One possible decent tradeoff is to trigger this check and the events only when the service gets the ip. This means we won't warn users if they have an invalid configuration, but I think it's a good compromise and better than generating too many events, probably somewhere around here https://github.com/metallb/metallb/blob/cd0b6fe9ae1375e0e696c40d040cfc6f86465e42/controller/main.go#L89-L90

(please leave also a comment with the reason why we are doing it there, as it feels a bit odd to have it in the middle there).